packages/css/src/components/progress-list/progress-list.scss

```diff
@@ -153,7 +344,138 @@
     border-inline-color: var(--ams-progress-list-step-current-connector-border-inline-color);
   }
 
-  .ams-progress-list__step:last-child & {
+  .ams-progress-list > .ams-progress-list__step:last-child:not(.ams-progress-list__step--has-sub-steps) & {
+    display: none;
+  }
+
+  @supports selector(:has(*)) {
+    .ams-progress-list > .ams-progress-list__step:last-child:not(:has(.ams-progress-list__sub-steps)) & {
+      display: none;
+    }
+  }
+}
+
+.ams-progress-list-sub-step__connector {
+  align-self: stretch;
+  background-color: var(--ams-progress-list-step-sub-steps-sub-step-connector-background-color);
+  border-inline-color: var(--ams-progress-list-step-sub-steps-sub-step-connector-border-inline-color);
+  border-inline-style: var(--ams-progress-list-step-sub-steps-sub-step-connector-border-inline-style);
+  border-inline-width: var(--ams-progress-list-step-sub-steps-sub-step-connector-border-inline-width);
+  grid-area: 1 / 1; // Overlaps the marker and connector.
+  z-index: 1;
+
+  .ams-progress-list__sub-step--current & {
+    border-inline-color: var(--ams-progress-list-step-sub-steps-sub-step-current-connector-border-inline-color);
+  }
+
+  .ams-progress-list__sub-step--completed & {
+    background-color: var(--ams-progress-list-step-sub-steps-sub-step-completed-connector-background-color);
+    border-inline-color: var(--ams-progress-list-step-sub-steps-sub-step-completed-connector-border-inline-color);
+    border-inline-style: var(--ams-progress-list-step-sub-steps-sub-step-completed-connector-border-inline-style);
+  }
+
+  .ams-progress-list--heading-2 & {
+    block-size: var(--ams-progress-list-heading-2-step-sub-steps-sub-step-marker-block-size);
+  }
+
+  .ams-progress-list--heading-3 & {
+    block-size: var(--ams-progress-list-heading-3-step-sub-steps-sub-step-marker-block-size);
+  }
+
+  .ams-progress-list--heading-4 & {
+    block-size: var(--ams-progress-list-heading-4-step-sub-steps-sub-step-marker-block-size);
+  }
+
+  // Style the connector when it belongs to the last sub-step of the last main step
+  // .ams-progress-list > .ams-progress-list__step:last-child → last main step
+  // .ams-progress-list__sub-steps → the sub-steps container within that step
+  // .ams-progress-list__sub-step:last-child → the last sub-step in that container
+  .ams-progress-list
+    > .ams-progress-list__step:last-child
+    .ams-progress-list__sub-steps
+    .ams-progress-list__sub-step:last-child
+    & {
+    border-inline-color: var(--ams-progress-list-step-sub-steps-sub-step-connector-last-step-border-inline-color);
+  }
+}
+
+.ams-progress-list-sub-step__connector--before {
```

Comment:
Why do we need this element? To overlay a dashed connector onto the solid one? Seems like a lot of code.

I was thinking, what if we made the connector of the parent step dashed if it’s the current one, then overlay solid connectors on top of that – so the other way around. Would that help simplify things?